### PR TITLE
cliconf and netconf refactor of iosxr_logging

### LINKED ISSUE #33324: [NETCONF] Refactor IOSXR modules to support netconf plugin
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME
iosxr

##### ANSIBLE VERSION

```
2.5
```

##### OS / ENVIRONMENT
macOS

##### SUMMARY
Refactor module codes to use netconf plugin:

- [x] iosxr_banner 
- [x] iosxr_logging
- [x] iosxr_interface
- [x] iosxr_system
- [x] iosxr_user
- [ ] ~iosxr_facts~

Note: iosxr_config, iosxr_command and iosxr_netconf shall support `cli` only.

### PR DESCRIPTION
**Pull request by kedarkekan**
##### SUMMARY

Fixes #33324 

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
iosxr_logging

##### ANSIBLE VERSION

```
2.5.0
```

### CODE REVIEW

**Comment by kedarkekan** [comment]
```
PLAY RECAP ****************************************************************************************************************
iosxr01                    : ok=118  changed=33   unreachable=0    failed=0

 56092 1515145568.29567: RUNNING CLEANUP
Command exited with status 0 after 368.212996006 seconds.
```

File: lib/ansible/modules/network/iosxr/iosxr_logging.py [hunk 1]
```diff
@@ -16,318 +16,673 @@
...
-      - Destination of the logs.
-    choices: ['on', 'hostnameprefix', console', 'monitor', 'buffered']
+      - Destination for system logging (syslog) messages.
+    choices: ['host', 'console', 'monitor', 'buffered', 'file']
   name:
     description:
-      - If value of C(dest) is I(file) it indicates file-name,
-        for I(user) it indicates username and for I(host) indicates
-        the host name to be notified.
+      - If value of C(dest) is:
```
Comment (gundalow) [inline lib/ansible/modules/network/iosxr/iosxr_logging.py]: I think this `:` is breaking `ansible-doc`

File: lib/ansible/modules/network/iosxr/iosxr_logging.py [hunk 2]
```diff
@@ -16,318 +16,673 @@
...
+      - vrf name when syslog server is configured, C(dest) = C(host)
+    default: default
+    version_added: 2.5
   size:
     description:
-      - Size of buffer. The acceptable value is in range from 307200 to
-        125000000 bytes.
-    default: 307200
+      - * Size of buffer when C(dest) = C(buffered). The acceptable value is in the range I(307200 to 125000000 bytes). Default: 307200
+        * Size of file when C(dest) = C(file). The acceptable value is in the range I(1 to 2097152)KB. Default: 2 GB
```
Comment (gundalow) [inline lib/ansible/modules/network/iosxr/iosxr_logging.py]: `Default: 2 GB` will cause issues, it's assumed to be `key: value`
You can quote the whole line

File: lib/ansible/modules/network/iosxr/iosxr_logging.py (hunk 1 above)
Comment (gundalow) [reply lib/ansible/modules/network/iosxr/iosxr_logging.py]: How about

```
description:
  - I(file) name indicates file-name.
  - I(host) name indicates the host-name or IP-address of syslog server.
```

File: lib/ansible/modules/network/iosxr/iosxr_logging.py [hunk 3]
```diff
@@ -343,30 +697,29 @@ def main():
...
     argument_spec.update(iosxr_argument_spec)
 
-    required_if = [('dest', 'hostnameprefix', ['name'])]
+    mutually_exclusive = [('dest', 'facility', 'hostnameprefix')]
+
+    required_if = [('dest', 'host', ['name']),
+                   ('dest', 'file', ['name']),
+                   ('dest', 'buffered', ['size', 'level']),
+                   ('dest', 'console', ['level']),
+                   ('dest', 'monitor', ['level'])]
```
Comment (ganeshrn) [inline lib/ansible/modules/network/iosxr/iosxr_logging.py]: Please add mutually_exclusive and required_if in the aggregate spec as well. This ensures checks are applied to the individual item in aggregate.
```
    argument_spec = dict(
        aggregate=dict(type='list', elements='dict', options=aggregate_spec, mutually_exclusive= mutually_exclusive, required_if=required_if),
    )
```

Comment (kedarX) [reply lib/ansible/modules/network/iosxr/iosxr_logging.py]: @ganeshrn you meant `self._module._check_required_if(required_if, item)` at line 223

**Review by trishnaguha — APPROVED** [review]

**Pull request merged**